src/core/thread/meshcop_dataset.cpp

```diff
@@ -41,15 +41,15 @@
 namespace Thread {
 namespace MeshCoP {
 
-Dataset::Dataset(void) :
+Dataset::Dataset(const uint8_t aType) :
+    mType(aType),
     mLength(0)
 {
-    mTimestamp.Init();
 }
 
 void Dataset::Clear(void)
 {
-    mTimestamp.Init();
+    mType = 0;
```

Comment:
0 indicates that this Dataset has been cleared, and we have not specified the type.
I suppose that type can either be kActiveTimestamp or kPendingTimestamp here?